pkg/cmd/experimental/syncgroups/sync-groups.go

```diff
@@ -0,0 +1,188 @@
+  $ %[1]s ldap all --sync-config=/path/to/ldap-sync-config.yaml
+
+  // Sync specific groups specified in a whitelist file with an LDAP server 
+  $ %[1]s ldap whitelist /path/to/whitelist.txt --sync-config=/path/to/ldap-sync-config.yaml
+
+  // Sync all OpenShift Groups that have already been synced with an LDAP server
+  $ %[1]s openshift all --sync-config=/path/to/ldap-sync-config.yaml
+
+  // Sync specific OpenShift Groups if they have already been synced with an LDAP server
+  $ %[1]s openshift whitelist groups/group1 groups/group2 groups/group3 --sync-config=/path/to/ldap-sync-config.yaml
+`
+)
+
+// GroupSyncScope determines the scope of the group sync operation
+type GroupSyncScope string
+
+const (
+	// GroupSyncScopeAll determines that all groups from the source should be synced
+	GroupSyncScopeAll GroupSyncScope = "All"
+	// GroupSyncScopeWhitelist determines that a whitelist of groups from the source should be synced
+	GroupSyncScopeWhitelist GroupSyncScope = "Whitelist"
+)
+
+func ValidateScope(scope GroupSyncScope) bool {
+	knownScopes := util.NewStringSet(string(GroupSyncScopeAll), string(GroupSyncScopeWhitelist))
+	return knownScopes.Has(string(scope))
+}
+
+// GroupSyncSource determines the source of the groups to be synced
+type GroupSyncSource string
+
+const (
+	// GroupSyncSourceLDAP determines that the groups to be synced are determined from an LDAP record
+	GroupSyncSourceLDAP GroupSyncSource = "LDAP"
+	// GroupSyncSourceOpenShift determines that the groups to be synced are determined from OpenShift records
+	GroupSyncSourceOpenShift GroupSyncSource = "OpenShift"
+)
+
+func ValidateSource(source GroupSyncSource) bool {
+	knownSources := util.NewStringSet(string(GroupSyncSourceLDAP), string(GroupSyncSourceOpenShift))
+	return knownSources.Has(string(source))
+}
+
+type SyncGroupsOptions struct {
+	// Source determines the source of the list of groups to sync
+	Source GroupSyncSource
+
+	// Scope determines the scope of the group sync
+	Scope GroupSyncScope
+
+	// ConfigSource is the path to the sync config
+	ConfigSource string
+
+	// WhitelistContents are the contents of the whitelist - either names of OpenShift groups given by
+	// an administrator on the command line invocation of the sync-groups command or LDAP group UIDs
+	// read from a whitelist file
+	WhitelistContents []string
+
+	// GroupsInterface is the interface used to interact with OpenShift Group objects
+	GroupInterface osclient.GroupInterface
+
+	// Stderr is the writer to write warnings and errors to
+	Stderr io.Writer
+
+	// Out is the writer to write output to
+	Out io.Writer
+
+	// Quiet controls whether warnings are output
+	Quiet bool
+}
+
+func NewSyncGroupsOptions() *SyncGroupsOptions {
+	return &SyncGroupsOptions{
+		Stderr:            os.Stderr,
+		WhitelistContents: []string{},
+	}
+}
+
+func NewCmdSyncGroups(name, fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	options := NewSyncGroupsOptions()
```

Comment:
Create a different struct for holding the flag values.  Pass that object into `Complete`